libraries/AP_SurfaceDistance/AP_SurfaceDistance.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <AP_Math/AP_Math.h>
+#include <Filter/LowPassFilter.h>
+#include <AP_InertialNav/AP_InertialNav.h>
+
+class AP_SurfaceDistance {
+public:
+    AP_SurfaceDistance(Rotation rot, const AP_InertialNav& inav, uint8_t i) : rotation(rot), inertial_nav(inav), instance(i) {};
+
+    void update();
+
+    // check if the last healthy range finder reading is too old to be considered valid
+    bool data_stale(void) const;
+
+    // helper to check that rangefinder was last reported as enabled and healthy
+    bool enabled_and_healthy(void) const;
+
+    // get inertially interpolated rangefinder height
+    bool get_rangefinder_height_interpolated_cm(int32_t& ret) const;
+
+    bool enabled;                          // Not to be confused with rangefinder enabled, this state is to be set by the vehicle.
+    bool alt_healthy;                      // true if we can trust the altitude from the rangefinder
+    int16_t alt_cm;                        // tilt compensated altitude (in cm) from rangefinder
+    float inertial_alt_cm;                 // inertial alt at time of last rangefinder sample
+    uint32_t last_healthy_ms;
+    LowPassFilterFloat alt_cm_filt {0.5};  // altitude filter
+    int16_t alt_cm_glitch_protected;       // last glitch protected altitude
+    int8_t glitch_count;                   // non-zero number indicates rangefinder is glitching
+    uint32_t glitch_cleared_ms;            // system time glitch cleared
+    float terrain_offset_cm;               // filtered terrain offset (e.g. terrain's height above EKF origin)
+
+    const AP_InertialNav& inertial_nav;
+    const Rotation rotation;
+
+private:
+
+    void Log_Write(void) const;
+
+    enum class Status {
+        Disabled = 0U,
+        Enabled = 1U<<0,
+        Unhealthy = 1U<<1,
+        Stale_Data = 1U<<3,
+        Glitch_Detected = 1U<<4,
+    };
+
+    // helper for setting status
+    void set_status(Status state_change);
+
+    const uint8_t instance;
+    uint8_t status;
```

Comment:
```suggestion
    Status status;
```

Reply:
Can't do bitmasks with enum class unless I define the | operator for the class.  As I couldn't find this workflow in the rest of the code I decided to do it as I have done it.  Open to other ideas though.